packages/adapter-cloudflare/index.js

```diff
@@ -62,29 +62,34 @@ export default function () {
 }
 
 /**
- * @param {string} app_dir
+ * @param {import('@sveltejs/kit').Builder} builder
  * @param {string[]} assets
  * @returns {import('.').RoutesJSONSpec}
  */
-function get_routes_json(app_dir, assets) {
+function get_routes_json(builder, assets) {
+	/**
+	 * The list of routes that will _not_ invoke functions (which cost money).
+	 * This is done on a best-effort basis, as there is a limit of 100 rules
+	 */
+	const exclude = [`/${builder.config.kit.appDir}/*`];
+
+	for (const file of assets) {
+		if (!file.startsWith(`${builder.config.kit.appDir}/`)) {
+			exclude.push(`/${file}`);
+		}
+	}
+
+	for (const path of builder.prerendered.paths) {
+		if (!builder.prerendered.redirects.has(path)) {
+			exclude.push(path);
+		}
+	}
+
 	return {
 		version: 1,
 		description: 'Generated by @sveltejs/adapter-cloudflare',
 		include: ['/*'],
-		exclude: [
-			`/${app_dir}/immutable/*`,
-			...assets
-				// We're being conservative by not excluding all assets in
-				// /static just yet. If there are any upstream auth rules to
-				// protect certain things (e.g. a PDF that requires auth),
-				// then we wouldn't want to prevent those requests from going
-				// to the user functions worker.
-				// We do want to show an example of a _routes.json that
-				// excludes more than just /_app/immutable/*, and favicons
-				// are a reasonable choice
-				.filter((file) => file.startsWith('favicon'))
-				.map((file) => `/${file}`)
-		]
+		exclude: exclude.slice(0, 99)
```

Comment:
`.slice(0, 99)`? Is there a limit of 100 items? We should at least log a warning if we hit that condition

Reply:
yep, there's a limit (left a comment a few lines earlier to explain it; the limits are documented [here](https://developers.cloudflare.com/pages/platform/functions/routing/#limits)). Added a warning